### PR TITLE
Use generic dictionary and list in ACNativeWindow

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Use generic dictionary and list in ACNativeWindow.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7001)

### CODE REVIEW

**Review by dreddy-work — APPROVED** [review]
LGTM

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ACNativeWindow.cs [hunk 1]
```diff
@@ -19,16 +18,15 @@ private sealed class ACNativeWindow : NativeWindow
         {
             internal static int s_inWndProcCnt;
 
-            // This hashtable can contain null for those ACWindows we find, but are sure are not ours.
-            private static readonly Hashtable s_ACWindows = new Hashtable();
+            // This dictionary can contain null for those ACWindows we find, but are sure are not ours.
+            private static readonly Dictionary<IntPtr, ACNativeWindow?> s_ACWindows = new Dictionary<IntPtr, ACNativeWindow?>();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ACNativeWindow.cs:22]: Consider:
```
 private static readonly Dictionary<IntPtr, ACNativeWindow> s_ACWindows = new();
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ACNativeWindow.cs [hunk 2]
```diff
@@ -112,12 +110,12 @@ internal static void RegisterACWindow(IntPtr acHandle, bool subclass)
             /// </summary>
             internal static void ClearNullACWindows()
             {
-                ArrayList nulllist = new ArrayList();
-                foreach (DictionaryEntry e in s_ACWindows)
+                List<IntPtr> nulllist = new List<IntPtr>();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ACNativeWindow.cs:113]: Consider:
```
List<IntPtr> toRemove = new();
```

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good, I added some nits.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ACNativeWindow.cs (hunk 2 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ACNativeWindow.cs]: suggestion was to rename `nulllist -> toRemove` or any other appropriate name.

**Pull request merged**